### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by gausie**
* Split out commands into separate modules
* Refactor most many commands to be consistent, DRY and less error-prone.
* General style improvements.

I span up this bot and tested it on a private server and it works great.

### CODE REVIEW

File: src/commands/misc/remind.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  if (timeMatcher.test(time)) {
+    const timeMatch = timeMatcher.exec(time);
+    timeToWait =
+      7 * 24 * 60 * 60 * 1000 * parseInt(timeMatch?.groups?.weeks || "0") +
+      24 * 60 * 60 * 1000 * parseInt(timeMatch?.groups?.days || "0") +
+      60 * 60 * 1000 * parseInt(timeMatch?.groups?.hours || "0") +
+      60 * 1000 * parseInt(timeMatch?.groups?.minutes || "0") +
+      1000 * parseInt(timeMatch?.groups?.seconds || "0");
+    reminderTime = Date.now() + timeToWait;
+    await interaction.reply(`Okay, I'll remind you in ${time}.`);
```
Comment (docrostov) [inline src/commands/misc/remind.ts:38]: i think there is a small issue where there is an effective max on the amount of time the system can actually store for your reminder; when you or someone else did a 20 year reminder it just never got pushed to the db. i assume it is literally just due to 9007199254740991 int limit. so maybe a note here of like "Reminder not added; it was too far in the future, I encourage you to live dangerously in the present." if it's over the int limit?

File: src/commands/misc/remind.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          }
+        }, reminder.reminder_time - now);
+      }
+    } else {
+      if (reminder.reminder_time - now < 7 * 24 * 60 * 60 * 1000) {
+        const user = await discordClient.client().users.fetch(reminder.user_id);
+        const channel = await user.createDM();
+        setTimeout(() => {
+          try {
+            (channel as DMChannel).send({
```
Comment (docrostov) [inline src/commands/misc/remind.ts:115]: thank you for this

**Review by docrostov — APPROVED** [review]
overall i like basically every structural change here and all my comments are minor shit that can be done in separate PRs and issues. i'm going to deploy this to oaf proper so we can test the commands you couldn't test and we can merge it once we confirm they work

File: src/things.ts [hunk 3]
```diff
@@ -1,11 +1,252 @@
...
+  ["trick-or-treating tot", "Restores your hp and mp after combat.\nHas varying abilities.\n"],
+  ["xiblaxian holo-companion", "Increases your combat initiative.\nStaggers enemies in combat.\n"],
+  ["black cat", "Is adorable.\nGenerally messes with you.\n"],
+  ["o.a.f.", "Is optimal.\nGenerally messes with you.\n"],
+]);
 
 export abstract class Thing {
   abstract name(): string;
-  abstract addToEmbed(embed: MessageEmbed, client: KOLClient): void;
+  abstract addToEmbed(embed: MessageEmbed, client: KoLClient): void;
```
Comment (horrible-little-slime) [inline src/things.ts:249]: In the abstract class, `addToEmbed` isnt' async

File: src/things.ts [hunk 4]
```diff
@@ -397,7 +631,7 @@ export class Item implements Thing {
     return `${description_string}${blueText}${autosell}${price_section}${zapGroup}${foldGroup}${container}${contents}`;
   }
 
-  async addToEmbed(embed: MessageEmbed, client: KOLClient): Promise<void> {
+  async addToEmbed(embed: MessageEmbed, client: KoLClient): Promise<void> {
```
Comment (horrible-little-slime) [inline src/things.ts:634]: in this implementation, `addToEmbed` is async

File: src/things.ts [hunk 5]
```diff
@@ -455,7 +689,7 @@ export class Effect implements Thing {
     return this._name;
   }
 
-  async addToEmbed(embed: MessageEmbed, client: KOLClient): Promise<void> {
+  async addToEmbed(embed: MessageEmbed, client: KoLClient): Promise<void> {
```
Comment (horrible-little-slime) [inline src/things.ts:692]: this one too

File: src/things.ts [hunk 6]
```diff
@@ -839,7 +1073,7 @@ export class Monster implements Thing {
     return description;
   }
 
-  async addToEmbed(embed: MessageEmbed, client: KOLClient): Promise<void> {
+  async addToEmbed(embed: MessageEmbed, client: KoLClient): Promise<void> {
```
Comment (horrible-little-slime) [inline src/things.ts:1076]: seems like they are all async

File: src/wikisearch.ts [hunk 7]
```diff
@@ -257,18 +362,18 @@ export class WikiSearcher {
...
-      text: "Problems? Message DocRostov#7004 on discord.",
-      iconURL: "http://images.kingdomofloathing.com/itemimages/oaf.gif",
-    });
+
+    const embed = createEmbed().setTitle(foundName.name).setURL(foundName.url);
+
     if (this._thingMap.has(foundName.name.toLowerCase())) {
       const thing = this._thingMap.get(foundName.name.toLowerCase());
-      await thing?.addToEmbed(embed, this._client);
+      thing?.addToEmbed(embed, this._client);
```
Comment (horrible-little-slime) [inline src/wikisearch.ts:370]: this still needs to be `await`ed

**Review by horrible-little-slime — CHANGES_REQUESTED** [review]

**Comment by docrostov** [comment]
would like us to figure out why /wiki worked on the other server but did not work on ASS before we merge up into main but this is currently actively deployed within ASS so we can merge into main whenever. we also should bump this guy over to loathers while we're messing with him and i can do a revision of the docs

**Comment by gausie** [comment]
The answer is that /wiki worked but [[wiki]] didn't

File: src/commands/clan/whitelist.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+import { ALL_CLANS } from "../../clans";
+import { KoLClient } from "../../kol";
+import { Command } from "../type";
+
+async function whitelist(interaction: CommandInteraction, kolClient: KoLClient): Promise<void> {
+  const roles = interaction.member?.roles as GuildMemberRoleManager;
+  if (
+    roles.cache.some(
+      (role) => role.id === "473316929768128512" || role.id === "466624206126448641"
```
[the remark this answers is not in the document]
Comment (gausie) [reply src/commands/clan/whitelist.ts:12]: it makes me laugh that we had a useless  constants.ts file that these weren't in

File: src/commands/kol/leprechaun.ts [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+function lep(interaction: CommandInteraction): void {
+  const weight = interaction.options.getInteger("weight", true);
+  if (weight <= 0) {
+    interaction.reply({ content: `Please supply a positive leprechaun weight.`, ephemeral: true });
+    return;
+  }
+
+  interaction.reply(
+    `A ${weight}lb leprechaun provides +${toDrop(weight).toFixed(2)}% meat drop. ` +
+      `(+${toDrop(weight, 1.25).toFixed(2)}% for Hobo Monkey)`
```
[the remark this answers is not in the document]
Comment (gausie) [reply src/commands/kol/leprechaun.ts:16]: that's fine, those functions already take a weight factor. It's annoying and confusing though that cookbb and vintner aren't weight factor they're just multiplicative of the outcome

File: src/commands/kol/spade.ts [hunk 10]
```diff
@@ -0,0 +1,309 @@
...
+// This is the maximum number of items we can have in our embeds
+const HORIZON = 25;
+
+enum SpadingFamiliars {
+  DEFAULT = 198,
+  GHOST = 282,
+}
+
+enum ItemType {
+  Food = "Food",
```
[the remark this answers is not in the document]
Comment (gausie) [reply src/commands/kol/spade.ts:17]: Nope!

**Pull request merged**